source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -533,12 +491,134 @@ namespace AdaptiveCards::Rendering::Uwp
+                                                                    bool isVisible,
+                                                                    bool isImageSvg,
+                                                                    S const& stream,
+                                                                    winrt::ImageSource const& imageSource) {
+        if (isImageSvg)
+        {
+            // If we have an SVG, we need to try to parse for the image size before setting the image source
+            auto streamSize = stream.Size();
+            auto inputStream = stream.GetInputStreamAt(0);
+            auto streamDataReader = winrt::DataReader(inputStream);
+            auto loadDataReaderOperation = streamDataReader.LoadAsync((uint32_t)streamSize);
+
+            loadDataReaderOperation.Completed(
+                [weakThis = this->get_weak(), streamDataReader, uiElement, isAutoSize, parentElement, imageContainer, isVisible, stream, imageSource](
+                    winrt::IAsyncOperation<uint32_t> const& result, winrt::AsyncStatus status) -> void
+                {
+                    if (auto strongThis = weakThis.get())
+                    {
+                        if (status == winrt::AsyncStatus::Completed)
+                        {
+                            auto bytes = result.GetResults();
+                            auto svgText = streamDataReader.ReadString(bytes);
+
+                            auto svgDocument = winrt::XmlDocument();
+                            svgDocument.LoadXml(svgText);
+
+                            strongThis->ParseXmlForHeightAndWidth(svgDocument, imageSource);
+                        }
+
+                        // Now that we've parsed the height, we can set the image source
+                        strongThis->SetSvgImageSourceAsync(
+                            imageSource.as<winrt::SvgImageSource>(), stream, uiElement, isAutoSize, parentElement, imageContainer, isVisible);
+                    }
+                });
+        }
+        else
+        {
+            auto setSourceAction = imageSource.as<winrt::BitmapImage>().SetSourceAsync(stream);
+
+            setSourceAction.Completed([weakThis = this->get_weak(), uiElement, isAutoSize, parentElement, imageContainer, isVisible](
+                winrt::IAsyncAction const& /*operation*/, winrt::AsyncStatus status)
+                {
+                    if (status == winrt::AsyncStatus::Completed && isAutoSize)
+                    {
+                        if (auto strongThis = weakThis.get())
+                        {
+                            strongThis->SetAutoSize(uiElement, parentElement, imageContainer, isVisible, false /* imageFiresOpenEvent */);
+                        }
+                    }
+                });
+        }
+    }
+
+    winrt::IAsyncAction render_xaml::XamlBuilder::SetSvgUriSource(winrt::SvgImageSource image, winrt::Uri imageUrl)
```

Comment:
```suggestion
    winrt::IAsyncAction render_xaml::XamlBuilder::SetSvgUriSource(winrt::SvgImageSource const& image, winrt::Uri const& imageUrl)
```
these 2 **must** be alive here so that the coroutine succeeds :)

Reply:
I switched all of the non-primitives to `weak_ref`. Let me know what you think :)

Reply:
I didn't mean to switch all of the non-primitives to `weak_ref` :)
only those that you're unsure will be alive by the time they're used in coroutines:)
good example of it is `XamlBuilder` which is passed by `weak_ref` around: it might be destroyed by the time it's called from a coroutine :)
I'd look at the objects, thought of scenarios where they might not survive, and if that's the case and it doesn't break the intended use of the object (may be it really have to be alive for everything to fall into place) - then use `weak_ref`.